couchdb-audit/log.js

```diff
@@ -139,18 +139,44 @@ module.exports = {
     }
 
     function get(docId, callback) {
-      getAll([docId], function(err, result) {
+      getAll([getAuditId(docId)], function(err, result) {
         callback(err, result && result.length && result[0]);
       });
     }
 
-    function getAll(docIds, callback) {
-      var ids = docIds.map(getAuditId);
-      auditDb.allDocs({ include_docs: true, keys: ids }, function(err, result) {
+    var MULTI_DOC_BATCH = 101;
+    function getAll(docIds, callback, results) {
+      results = results || {rows: []};
+
+      if (!docIds || !docIds.length) {
+        return callback(null, results);
+      }
+
+      var batch = docIds.splice(docIds.length - MULTI_DOC_BATCH);
+      auditDb.allDocs({ include_docs: true, keys: batch }, function(err, batchResult) {
+        if (err) {
+          return callback(err);
+        }
+
+        results.rows = results.rows.concat(batchResult.rows);
+
+        getAll(docIds, callback, results);
```

Comment:
Happy to re-work it to use `async` (I'll look at the docs, but still like `whilst` probably work here).

FWIW I'm not seeing any massive memory inefficiency: it's all pass by reference, so at most you have a few layers of array metadata you wouldn't otherwise have. Without looking at and nit picking the implementation I also don't think it would be a given that async would be necessarily more efficient in any way that relevant. Having said that, if people find async easier to read we should definitely go with that.